ibm_vpd_app.cpp

```diff
@@ -127,11 +127,23 @@ static auto isFruInVpdCache(const std::string& objectPath)
  */
 static auto needsRecollection(const nlohmann::json& js, const string& filePath)
 {
-    if (js["frus"][filePath].at(0).value("pluggableAtStandby", false) ||
-        js["frus"][filePath].at(0).value("concurrentlyMaintainable", false))
+    if (js["frus"][filePath].at(0).value("pluggableAtStandby", false))
     {
         return true;
     }
+
+    const std::vector<nlohmann::json>& groupEEPROM =
+        js["frus"][filePath].get_ref<const nlohmann::json::array_t&>();
+
+    for (const auto& item : groupEEPROM)
+    {
+        if (item.find("concurrentlyMaintainable") != item.end() &&
+            (item.value("concurrentlyMaintainable", false)))
```

Comment:
(item.find("concurrentlyMaintainable") != item.end()) this can be removed. 
As item.value() returns default if the given key doesn't exist. Can you please check and make changes as needed?

Reply:
Done